### PR TITLE
Allow to instrument client pool sizes

### PR DESCRIPTION
**Pull request by dim**

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -61,6 +61,20 @@ func (c *ClusterClient) Close() error {
 	return nil
 }
 
+// PoolLen returns the number of active connections in the pool, for instrumentation
```
Comment (vmihailenco) [inline cluster.go]: What do you mean by "instrumentation"?

Also description is not very accurate since you return max number of active connections...

Comment (dim) [inline cluster.go]: I would like measure and periodically report the actual number of used pool connections

Comment (vmihailenco) [inline cluster.go]: Are you sure that you don't want to return the sum? If not - please update the comment. Your implementation is *very* unusual and specific...

Comment (dim) [inline cluster.go]: So it all goes back to the fact that `ClusterOptions` accept a `PoolSize` which is actually per-client and not a per-cluster-client. Returning the `PoolLen()` is supposed to signal the user that the `PoolSize` setting is probably too low, therefore using the maximum seems the right approach to me, I will update the comment.

**Comment by vmihailenco** [comment]
LGTM, thanks.

Let me know if you find that default limit of 10 connections is not enough...

**Comment by vmihailenco** [comment]
...and feel free to merge this and release new version :)

**Comment by dim** [comment]
Closing in favour of https://github.com/go-redis/redis/pull/241

**Pull request closed without merging**